2 and 3 homework

SleeplessShu opened the pull request:
Домашняя работа.
созданы: 
-стартовый экран
-настройки
-медиатека
-поиск

добавлена кнопка "назад"

SleeplessShu committed:
Изменена тема с Material3 на MaterialComponents

SleeplessShu committed:
Изменена тема с Material3 на MaterialComponents в night themes

SleeplessShu committed:
Домашнее задание 2 и 3:
Добавлены экраны
-основной
-настройки
-медиатека
-поиск

faritowich commented on app/src/main/java/com/practicum/playlistmaker_ver2/MediatekaActivity.kt:
```diff
@@ -0,0 +1,20 @@
+        val reactionOnButtonBackToMain = findViewById<ImageView>(R.id.buttonBackToMain)
+        reactionOnButtonBackToMain.setOnClickListener {
+            val intentBackToMain = Intent(this, MainActivity::class.java)
+            startActivity(intentBackToMain)
```
Для перехода назад надо использовать не интент, а методы Activity finish() или onBackPressed(), потому что иначе вместо возврата в предыдущий экран создается новый экземпляр MainActivity и при возврате обратно на главный экран и нажатии на кнопку телефона "назад" мы "вернемся" снова на этот экран. На эту тему можно почитать про стек активити. Это будет критично в следующем спринте. Ниже аналогично

SleeplessShu replied:
Спасибо, мне кажется, я опередил паровоз. Про стек активити почитал, возвраты сделал через finish()

faritowich commented on app/src/main/java/com/practicum/playlistmaker_ver2/MainActivity.kt:
```diff
@@ -1,11 +1,38 @@
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        val reactionOnButtonSettings = findViewById<Button>(R.id.buttonSettings)
```
Название неоправданно длинное, имхо)

SleeplessShu replied:
заменил на onPressSettings и аналоги по всему проекту

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,84 @@
+
+            android:backgroundTint="#ffffff"
+            android:text="@string/Search"
+            android:textSize="22dp"
```
Для размеров шрифтов всегда используется sp, а не dp

SleeplessShu replied:
это моя невнимательность. исправил по всему коду, заменив на вынесенную в values константу

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,84 @@
+            app:iconGravity="textStart"
+            app:iconPadding="8dp"
+            app:iconTint="@color/black"
+            android:foreground="?attr/selectableItemBackground" />/>
```
Лишняя закрывашка в конце

SleeplessShu replied:
убрал

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,142 @@
+            android:textColor="@color/black"
+            android:textSize="22sp" />
+
+    </LinearLayout>
```
Для тулбара можно также использовать один TextView и задавать иконку при помощи атрибута drawableStart или использовать androidx.appcompat.widget.Toolbar

SleeplessShu replied:
реализовал этот вариант в шапке настроек для эксперимента.

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,142 @@
+            android:layout_height="wrap_content"
+            android:orientation="horizontal"
+            android:layout_gravity="center"
+            android:paddingVertical="21dp"
```
Рекомендуется делать размеры View и отступы кратными 8 (или 4). По крайней мере чётными. Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

SleeplessShu replied:
Переделал размеры на константы, кратные 8

SleeplessShu committed:
Домашнее задание 2 и 3
внесены правки:
- цвета перенесены в values
- размеры перенесены в values
- лишняя скобка убрана
- возврат в главное меню реализован через finish()
- размер шрифтов sp, а не dp

- в шапке SettingsActivity иконка и кнопка слиты в <TextView>
